src/change_stream.ts

```diff
@@ -726,6 +726,16 @@ export class ChangeStream<
     }, callback);
   }
 
+  async *[Symbol.asyncIterator](): AsyncIterator<TChange, void> {
+    if (this.closed) {
+      return;
+    }
+
+    while (await this.hasNext()) {
```

Comment:
sorry I didn't consider this earlier, but I think we should add a `try-finally` block here.  We don't need the `catch`, because we want to propagate the error, but in the `finally`, we should close the change stream.  This would cleanup the change stream automatically if a user `break`s out of the for-await loop.

This introduces some extra test cases as well

- when there is an error thrown, we should confirm that the change stream is closed
- when if `return` is called on the async iterator, then the change stream is closed

Reply:
Good catch, makes total sense.